internal/sql/driver/sql_service.go

```diff
@@ -0,0 +1,199 @@
+	cb                   *cb.CircuitBreaker
+}
+
+func newSQLService(cm *cluster.ConnectionManager, ss *iserialization.Service, fac *cluster.ConnectionInvocationFactory, is *invocation.Service) *SQLService {
+	cbr := cb.NewCircuitBreaker(
+		cb.MaxRetries(math.MaxInt32),
+		cb.RetryPolicy(func(attempt int) time.Duration {
+			return time.Duration((attempt+1)*100) * time.Millisecond
+		}),
+	)
+	return &SQLService{
+		connectionManager:    cm,
+		serializationService: ss,
+		invFactory:           fac,
+		invService:           is,
+		cb:                   cbr,
+	}
+}
+
+// ExecuteSQL runs the given SQL query on the member-side.
+// Placeholders in the query is replaced by params.
+// A placeholder is the question mark (?) character.
+// For each placeholder, a corresponding param should exist.
+func (s *SQLService) ExecuteSQL(ctx context.Context, query string, params []driver.Value) (*ExecResult, error) {
+	cbs := ExtractCursorBufferSize(ctx)
+	tom := ExtractTimeoutMillis(ctx)
+	resp, err := s.executeSQL(ctx, query, expectedResultUpdateCount, tom, cbs, params)
+	if err != nil {
+		return nil, err
+	}
+	return resp.(*ExecResult), nil
+}
+
+// QuerySQL runs the given SQL query on the member-side.
+// Placeholders in the query is replaced by params.
+// A placeholder is the question mark (?) character.
+// For each placeholder, a corresponding parameter should exist.
+func (s *SQLService) QuerySQL(ctx context.Context, query string, params []driver.Value) (*QueryResult, error) {
+	cbs := ExtractCursorBufferSize(ctx)
+	tom := ExtractTimeoutMillis(ctx)
+	resp, err := s.executeSQL(ctx, query, expectedResultRows, tom, cbs, params)
+	if err != nil {
+		return nil, err
+	}
+	return resp.(*QueryResult), nil
+}
+
+func (s *SQLService) Fetch(qid isql.QueryID, conn *cluster.Connection, cbs int32) (*isql.Page, error) {
+	req := codec.EncodeSqlFetchRequest(qid, cbs)
+	resp, err := s.invokeOnConnection(context.Background(), req, conn)
+	if err != nil {
+		return nil, err
+	}
+	page, err := codec.DecodeSqlFetchResponse(resp, s.serializationService)
+	if err != (*isql.Error)(nil) {
+		return nil, err
+	}
+	return page, nil
+}
+
+func (s *SQLService) CloseQuery(qid isql.QueryID, conn *cluster.Connection) error {
+	req := codec.EncodeSqlCloseRequest(qid)
+	_, err := s.invokeOnConnection(context.Background(), req, conn)
+	if err != nil {
+		return fmt.Errorf("closing query: %w", err)
+	}
+	return nil
+}
+
+func (s *SQLService) executeSQL(ctx context.Context, query string, resultType byte, timeoutMillis int64, cursorBufferSize int32, params []driver.Value) (interface{}, error) {
+	serParams, err := s.serializeParams(params)
+	if err != nil {
+		return nil, err
+	}
+	conn := s.connectionManager.RandomConnection()
+	if conn == nil {
+		return nil, ihzerrors.NewIOError("no connection found", nil)
+	}
+	qid := isql.NewQueryIDFromUUID(conn.MemberUUID())
+	req := codec.EncodeSqlExecuteRequest(query, serParams, timeoutMillis, cursorBufferSize, "", resultType, qid)
```

Comment:
Is there a reason we don't support `schema` option? I guess it is not implemented on the server yet but not sure

Reply:
That was my impression as well. I did not include it since it is not supported yet on the server.